src/routes/club/Proposal.js

```diff
@@ -28,14 +34,80 @@ router.get("/get/:id", verifyClub, async (req, res) => {
 
 
 router.post("/create", verifyClub, async (req, res) => {
+	const newProposal = new Proposal(req.body);
 
-});
+	// TODO: Do post saving operations
+	Club.findById(req.session.userId)
+		.then(club => {
+			// Get club details
+			newProposal.clubName = club.name
+			newProposal.clubId = mongoose.Types.ObjectId(club)
+
+			// Save proposal to database
+			newProposal.save()
+				.then((proposal) => {
+					res.status(200).send({ success: true, id: proposal.id })
+				})
+				.catch((err) => {
+					return res.status(400).send({ success: false, message: "Data validation failed", error: err });
+				})
+		})
+		.catch((err) => {
+			res.status(404).send({ success: false, message: "Cannot find club ID" })
+		});
+
+})
 
 router.put("/edit/:id", verifyClub, async (req, res) => {
+	// TODO: Add a check so club cannot edit major or any data once it's approved by faculty
```

Comment:
I think that clubs should be able to store incomplete proposals which are not forwarded to any faculty so they can work on it over time. We shouldn't allow clubs to edit proposals at all after forwarding.

Reply:
Agreed. Will add a approval == 0 check. Maybe we can also add a submit or ready for approval field